qa/sc_evm_bootstrap.py

```diff
@@ -0,0 +1,253 @@
+        # Amount should be expressed in zennies
+        transferred_amount = Decimal(12.34)
+        transferred_amount_in_zennies = convertZenToZennies(transferred_amount)
+        transferred_amount_in_wei = convertZenniesToWei(transferred_amount_in_zennies)
+
+        recipient_keys = generate_account_proposition("seed3", 1)[0]
+        recipient_proposition = recipient_keys.proposition
+        logging.info("Trying to send {} zen to address {}".format(transferred_amount, recipient_proposition))
+
+        j = {
+            "from": evm_address,
+            "to": recipient_proposition,
+            "value": transferred_amount_in_zennies
+        }
+        request = json.dumps(j)
+        response = sc_node_1.transaction_sendCoinsToAddress(request)
+        logging.info("tx sent:")
+        logging.info(response)
+        self.sc_sync_all()
+
+        # get mempool contents
+        response_1 = sc_node_1.transaction_allTransactions()
+        response_2 = sc_node_2.transaction_allTransactions()
+        logging.info("mempool contents:")
+        logging.info(response_1)
+        assert_equal(response_1, response_2)
+
+        # tx json repr has amount in wei
+        tx_amount_in_wei = response_2["result"]["transactions"][0]["value"]
+        assert_equal(str(tx_amount_in_wei), str(transferred_amount_in_wei))
+
+        # send more zen with another from address to have more than one transaction in block
+        logging.info("Trying to send {} zen to address {}".format(transferred_amount, recipient_proposition))
+
+        j = {
+            "from": evm_address_2,
+            "to": recipient_proposition,
+            "value": transferred_amount_in_zennies
+        }
+        request = json.dumps(j)
+        response = sc_node_1.transaction_sendCoinsToAddress(request)
+        logging.info("tx sent:")
+        tx_hash_1 = response['result']['transactionId']
+        self.sc_sync_all()
+        tx_hash = response['result']['transactionId']
+
+        # request chainId via rpc route
+        logging.info("rpc response:")
+        logging.info(sc_node_1.rpc_eth_chainId())
+
+        # request getBalance via rpc route
+        logging.info("rpc response:")
+        logging.info(sc_node_1.rpc_eth_getBalance(evm_hex_address, "1"))
+
+        generate_next_blocks(sc_node_1, "first node", 1)
+        self.sc_sync_all()
+        sc_best_block = sc_node_1.block_best()["result"]
+        logging.info(sc_best_block)
+
+        # check if header contains correct gasUsed (2 * eoa to eoa transfer gas costs)
+        assert_equal(21000*2, sc_best_block['block']['header']['gasUsed'])
+
+        assert_equal(669921875, sc_best_block['block']['header']['baseFee'])
+        transactionFee, _, _ = computeForgedTxFee(sc_node_1, tx_hash)
+
+        # test eth_estimateGas with ERC20 (TestERC20.sol with additional mint function) contract bytecode
+        request = {
+            "from": "0x" + evm_address,
```

Comment:
This a very common cause for issues and an annoying one at that. The RPC spec is very clear on how the formatting must be and it includes the `0x` prefix. Our http api unfortunately does not accept the prefix so we have to constantly add or remove the prefix everywhere. We should refactor this, I see the following options:
- Make our http api accept the `0x` prefix too so we can just use it everywhere.
- Extend the request layer in the STF to automatically add or remove the prefixes depending on which api is called (http or json-rpc)